core/types/transaction.go

```diff
@@ -563,7 +572,7 @@ func (t *TransactionsByPriceAndNonce) Peek() *Transaction {
 func (t *TransactionsByPriceAndNonce) Shift() {
 	acc, _ := Sender(t.signer, t.heads[0].Tx)
 	if txs, ok := t.txs[acc]; ok && len(txs) > 0 {
-		if wrapped, err := NewTxWithMinerFee(txs[0], t.baseFee); err == nil {
+		if wrapped, err := NewTxWithMinerFee(txs[0], t.baseFee, acc); err == nil {
```

Comment:
I think we can get rid of this change and use this in the push gossiper instead: https://github.com/ethereum/go-ethereum/blob/master/core/types/transaction.go#L507